lib/Cake/Test/Case/Model/Behavior/TranslateBehaviorTest.php

```diff
@@ -1443,4 +1443,164 @@ public function testNoExtraRowsForAssociatedTranslations() {
 		$this->assertEquals('name', $results[0]['TranslateTestModel']['field']);
 	}
 
+	public function testBeforeFindAllI18nConditions() {
+		$this->loadFixtures('TranslateArticle', 'TranslatedArticle', 'User');
+		$TestModel = new TranslatedArticle();
+		$TestModel->cacheQueries = false;
+		$TestModel->locale = 'eng';
+		$expected = array(
+			'conditions' => array(
+				'NOT' => array('I18n__title.content' => ''),
+			),
+			'fields' => null,
+			'joins' => array(
+				array(
+					'type' => 'INNER',
+					'alias' => 'I18n__title',
+					'table' => (object)array(
+						'tablePrefix' => '',
+						'table' => 'article_i18n',
+						'schemaName' => 'cakephp_test',
+					),
+					'conditions' => array(
+						'TranslatedArticle.id' => (object)array(
+							'type' => 'identifier',
+							'value' => 'I18n__title.foreign_key',
+						),
+						'I18n__title.model' => 'TranslatedArticle',
+						'I18n__title.field' => 'title',
+						'I18n__title.locale' => 'eng',
+					),
+				),
+				array(
+					'type' => 'INNER',
+					'alias' => 'I18n__body',
+					'table' => (object)array(
+						'tablePrefix' => '',
+						'table' => 'article_i18n',
+						'schemaName' => 'cakephp_test',
+					),
+					'conditions' => array(
+						'TranslatedArticle.id' => (object)array(
+							'type' => 'identifier',
+							'value' => 'I18n__body.foreign_key',
+						),
+						'I18n__body.model' => 'TranslatedArticle',
+						'I18n__body.field' => 'body',
+						'I18n__body.locale' => 'eng',
+					),
+				),
+			),
+			'limit' => 2,
+			'offset' => null,
+			'order' => array(
+				'TranslatedArticle.id' => 'ASC',
+			),
+			'page' => 1,
+			'group' => null,
+			'callbacks' => true,
+			'recursive' => 0,
+		);
+		$query = array(
+			'conditions' => array(
+				'NOT' => array(
+					'I18n__title.content' => '',
+				),
+			),
+			'fields' => null,
+			'joins' => array(),
+			'limit' => 2,
+			'offset' => null,
+			'order' => array(
+				'TranslatedArticle.id' => 'ASC',
+			),
+			'page' => 1,
+			'group' => null,
+			'callbacks' => true,
+			'recursive' => 0,
+		);
+		$TranslateBehavior = ClassRegistry::getObject('TranslateBehavior');
+		$result = $TranslateBehavior->beforeFind($TestModel, $query);
+		$this->assertEquals($expected, $result);
+	}
+
+	public function testBeforeFindCountI18nConditions() {
+		$this->loadFixtures('TranslateArticle', 'TranslatedArticle', 'User');
+		$TestModel = new TranslatedArticle();
+		$TestModel->cacheQueries = false;
+		$TestModel->locale = 'eng';
+		$expected = array(
+			'conditions' => array(
+				'NOT' => array('I18n__title.content' => ''),
```

Comment:
I am not sure about that. TranslateArticleModel alias exists here only when `find('count'` is called. It does not exist here when `find('all'` is called. The join array with TranslateArticleModel is a bit suspicious because the join condition does not have `TranslateArticleModel.title` column. So it is not fully clear what records are joined by that when a model has more than one translatable fields.

Altering the query conditions only in `find('count'` will introduce more inconsistencies to the code comparing it with `find('all'`. That does not seems to be the right way to go. Altering the query conditions in both `find('count'` and  `find('all'` will require more changes because in that case the joins in `find('all'` will need to be altered to make TranslateArticleModel available. Perhaps it would be better to open another issue to make the behavior of `find('count'` and `find('all'` consistent using the same joins and altering conditions in the same way when necessary.